controllers/city.controller.go

```diff
@@ -0,0 +1,180 @@
+package controllers
+
+import (
+	"errors"
+	"github.com/gofiber/fiber/v2"
+	"net/http"
+	"project_weather/models"
+	"project_weather/repositories"
+)
+
+type CityDB interface {
+	FindCityByID(id string) (*models.City, error)
+	RegisterCity(city *models.City) error
+	UpdateCity(city *models.City) error
+	DeleteCityByID(id string) error
+	GetAllCity() ([]models.City, error)
+}
+
+type CityController struct {
+	DB repositories.CityRepo
+}
+
+func (r *CityController) GetCityById(ctx *fiber.Ctx) error {
+	id := ctx.Params("id")
+	if id == "" {
+		ctx.Status(http.StatusInternalServerError).JSON(&fiber.Map{
+			"status":  "error",
+			"message": "Id can't be empty",
+		})
+		return errors.New("Id can't be empty")
+	}
+
+	city, err := r.DB.FindCityByID(id)
+	if err != nil {
+		ctx.Status(http.StatusBadRequest).JSON(&fiber.Map{
+			"status":  "error",
+			"message": "City not found",
+		})
+		return err
+	}
+	ctx.Status(http.StatusOK).JSON(&fiber.Map{
+		"status":  "success",
+		"message": "City found",
+		"data":    city,
+	})
+	return nil
+}
+
+func (r *CityController) DeleteCity(ctx *fiber.Ctx) error {
+	id := ctx.Params("id")
+	city, err := r.DB.FindCityByID(id)
+
+	if err != nil {
+		ctx.Status(http.StatusBadRequest).JSON(&fiber.Map{
+			"status":  "error",
+			"message": "City not found",
+		})
+	}
+
+	if id == "" {
+		ctx.Status(http.StatusInternalServerError).JSON(&fiber.Map{
+			"status":  "error",
+			"message": "Invalid id",
+		})
+		return errors.New("Invalid id")
+	}
+	err = r.DB.DeleteCityByID(id)
+	if err != nil {
+		ctx.Status(http.StatusBadRequest).JSON(&fiber.Map{
+			"status":  "error",
+			"message": err,
+		})
+		return err
+	}
+	ctx.Status(http.StatusOK).JSON(&fiber.Map{
+		"status":  "success",
+		"message": "City deleted",
+		"data":    city,
+	})
+	return nil
+}
+
+func (r *CityController) RegisterCity(ctx *fiber.Ctx) error {
+	city := &models.City{}
+
+	err := ctx.BodyParser(&city)
+	if err != nil {
+		ctx.Status(http.StatusUnprocessableEntity).JSON(&fiber.Map{
```

Comment:
This should be status bad request. I think StatusUnprocessableEntity means something else [https://www.rfc-editor.org/rfc/rfc9110.html#name-422-unprocessable-content](https://www.rfc-editor.org/rfc/rfc9110.html#name-422-unprocessable-content)